apps/hash-api/src/graphql/resolvers/index.ts

```diff
@@ -111,6 +112,7 @@ export const resolvers: Omit<Resolvers, "Query" | "Mutation"> & {
     pageComments: loggedInAndSignedUpMiddleware(pageCommentsResolver),
     blocks: loggedInAndSignedUpMiddleware(blocksResolver),
     getEntity: getEntityResolver,
+    getFlowRuns,
```

Comment:
You're likely aware, but just realised we shouldn't merge this PR as is because it would expose the inputs/outputs of flow runs to any non-authenticated user. Perhaps we should add some check on getting just the users workflow runs or something like that, and until then block any response in the resolver when running in `production`.

Reply:
I've added a throw in prod in https://github.com/hashintel/hash/pull/4218/commits/39f02751246394fdc3de207f2ff73557a76c431c

We'll be updating how this works anyway so don't think there's any point in adding more at this stage.

We do need to decide which of the following we're doing though:
1. Get all flow runs a user has access to, and then request runs with those ids from Temporal
2. Somehow encode the information we need in Temporal (e.g. userId, webId) so we can filter on it

Assuming performance of 'get me all flows with ids in [giant list]' is fine, it's probably to do (1) since it means we don't have to reimplement the permission system in Temporal.

Reply:
Agreed, thank you